lib/python/bind_data_access.h

```diff
@@ -196,6 +203,22 @@ template <class... Ts> class as_ElementArrayViewImpl {
                          elems.dims().labels().end());
       std::iter_swap(labels.end() - 2, labels.end() - 1);
       return transpose(elems, labels);
+    } else if (view.dtype() == dtype<scipp::core::Quaternion>) {
+      return get_data_variable(view)
+          .template elements<scipp::core::Quaternion>();
+    } else if (view.dtype() == dtype<scipp::core::Translation>) {
+      return get_data_variable(view)
+          .template elements<scipp::core::Translation>();
+    } else if (view.dtype() == dtype<Eigen::Affine3d>) {
+      auto elems = get_data_variable(view).template elements<Eigen::Affine3d>();
+      elems = fold(
+          elems, Dim::InternalStructureComponent,
+          Dimensions({Dim::InternalStructureRow, Dim::InternalStructureColumn},
+                     {4, 4}));
+      std::vector labels(elems.dims().labels().begin(),
+                         elems.dims().labels().end());
+      std::iter_swap(labels.end() - 2, labels.end() - 1);
+      return transpose(elems, labels);
```

Comment:
Could be moved into a helper function, taking the matrix size as a parameter? Apart from the size this looks like a 1:1 duplication?